### ISSUE TITLE
[Feature Request] Modular DAG

### ISSUE THREAD

**Issue by GhisF** [dagu-dev/dagu#373]
It is not uncommon to have several versions of a DAG with only minor variations. It could be interesting to have the possibility to import steps or call other DAGs to avoid copy/paste between dags.

**Comment by yohamta** [comment]
Thanks for the great suggestion. Yes, as you mentioned there're two different approaches to dealing with the issue.
I think calling other DAGs (or sub-DAGs) natively looks like a good idea.

**Comment by ccxuy** [comment]
any updates?

**Comment by yohamta** [comment]
Not yet. While importing steps or calling other DAGs to avoid copy/paste between DAGs is an interesting idea, I think that allowing the direct importing of other DAGs in YAML could lead to some complications and potential issues with maintainability.

As an alternative, you can currently call other DAGs (or sub-DAGs) using the `dagu start` command.

We can continue to explore ways to improve the reusability and modularity of DAGs. Any suggestion or ideas are very welcomed.

**Comment by yohamta** [comment]
my idea:
```
steps:
  - sub_workflow1
     run: path/to/sub_workflow.yaml
     params: "A=1 B=2"
```

**Comment by GhisF** [comment]
It looks like a good idea.

**Comment by yohamta** [comment]
This feature is implemented in v1.13.0.
